File: LeetCode/C/206-linked-list-反转链表.cpp
/*
 * @lc app=leetcode.cn id=206 lang=cpp
 *
 * [206] 反转链表
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    ListNode *reverseList(ListNode *head) {
        ListNode *pre = nullptr;
        ListNode *cur = head;
        ListNode *next = head;
        while (cur != nullptr) {
            next = cur->next;
            cur->next = pre;
            pre = cur;
            cur = next;
        }
        return pre;
    }
};
// @lc code=end

File: LeetCode/Go/206-linked-list-反转链表.go
/*
 * @lc app=leetcode.cn id=206 lang=golang
 *
 * [206] 反转链表
 */
package leetcode

// @lc code=start
func reverseList(head *ListNode) *ListNode {
	var pre, cur, t *ListNode = nil, head, nil
	for cur != nil {
		t = cur.Next
		cur.Next = pre
		pre = cur
		cur = t
	}
	return pre
}

// @lc code=end
